tidy3d/components/device/sim_data.py

```diff
@@ -128,22 +129,36 @@ def plot_field(
         """
 
         monitor_data = self[monitor_name]
+        property_to_plot = None
+
+        if isinstance(monitor_data, TemperatureData):
```

Comment:
these `if/else` and the below ones for `field_name` seem to me that they'd be better organized being defined in Temperatureata and VoltageData? as methods `get_plot_propety( ... ) -> str` and `get_field_name(val: str)`

(we probably should do some of this in SimulationData as well ... eventually)